### ISSUE TITLE
Column becomes hidden after refreshing page ember-light-table@2.0.0-beta.2

### ISSUE THREAD

**Issue by ReasonableDeveloper** [offirgolan/ember-light-table#669]
Hey, first of all thank you for all the work on this library.
I'm using following versions that might be related to the issue.
```
"ember-cli": "~3.5.1",
"ember-light-table": "~2.0.0-beta.2",
"ember-responsive": "~3.0.0",
```
I'm having trouble with a column whose default is `hidden: true`. I'm then conditionally setting it to `hidden: false` to display it. This works fine but as soon as I hit refresh, the column is no longer displayed even though I'm still rendering the column.

I'm creating a table like this
```JavaScript
let table = new Table(this.get('columns'), this.model, {
  enableSync: true
})

this.set('table', table)
```

Printing `this.get('columns')` yields me following column. Please pay attention to the 'hidden property'.
```
[
  {
    align: "left"
    breakpoints: (2) ["desktop", "jumbo"]
    cellComponent: "foo-comp"
    draggable: true
    hidden: false
    isDebug: false
    label: "foo"
    minResizeWidth: 75
    resizable: false
    searchable: false
    sortable: false
    valuePath: "id"
    width: "40px"
  },
]
```

Afterwards printing `this.table.get('allColumns')` somehow shows that the hidden property is set to true.
```
[
  {
    align: "left"
    breakpoints: (2) ["desktop", "jumbo"]
    cellComponent: "foo-comp"
    columnId: (...)
    draggable: true
    hidden: true
    isDebug: false
    isGroupColumn: (...)
    isHidden: true
    isVisibleGroupColumn: (...)
    label: "foo"
    minResizeWidth: 75
    resizable: false
    responsiveHidden: (...)
    searchable: false
    sortable: false
    subColumns: (...)
    valuePath: "id"
    visibleSubColumns: (...)
    width: "40px"
  },
]
```
Since my conditional (that sets `hidden` to `true` or `false`) is based on the resolved value of a promise, I thought that I'm not awaiting the promise correctly but comparing `this.table.get('allColumns')` with `this.get('columns)` revealed the discrepancy between what I'm passing in to the constructor and what actually gets rendered.

**Comment by alexander-alvarez** [comment]
hey @ReasonableDeveloper  sorry for the delay in response (was on vacation).

that seems weird. are you creating `Column` instances or just passing in POJOs?

**Comment by ReasonableDeveloper** [comment]
The columns I'm passing to the `Table` constructor are plain old JavaScript
objects

**Comment by alexander-alvarez** [comment]
If you could send a link my way of a reproduction that I can debug, I'd be more than happy to work through this issue with you

**Comment by ReasonableDeveloper** [comment]
I tried to reproduce this and while doing so, I figured that the issue was caused by me. I was only initializing the table in the `init` of my `component` and did not mutate the table correctly for the template to pick up the changes and re-render.
I now have a computed `table` property that is updating the `hidden` column according to my needs and it works just as intented. :fireworks:

Thank you for your response. :+1:
